server/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -263,17 +262,14 @@ protected Query doToQuery(SearchExecutionContext context) throws IOException {
             );
         }
 
-        ObjectMapper nestedObjectMapper = context.getObjectMapper(path);
-        if (nestedObjectMapper == null) {
+        NestedObjectMapper mapper = context.nestedLookup().getNestedMappers().get(path);
+        if (mapper == null) {
             if (ignoreUnmapped) {
                 return new MatchNoDocsQuery();
             } else {
                 throw new IllegalStateException("[" + NAME + "] failed to find nested object under path [" + path + "]");
             }
         }
-        if (nestedObjectMapper.isNested() == false) {
-            throw new IllegalStateException("[" + NAME + "] nested object under path [" + path + "] is not of nested type");
-        }
```

Comment:
I am assuming this is checked elsewhere now, or is this exception no longer needed?